Document / vs // (#5960)
oscar-system/Oscar.jl: docs/src/Rings/intro.md

- orders in number fields
- series rings

## [Subtle but important: `/` vs `//`](@id subtle_distinction_for_rings)

In OSCAR, the operators `/` and `//` have distinct meanings.

- `x / y` performs exact division within the current algebraic structure, if defined.
- `x // y` constructs a formal quotient, placing the result in a fraction-field parent.

Example:

```julia

> julia> R, x = polynomial_ring(QQ, "x")

Reviewer: ```suggestion
julia> R, x = polynomial_ring(QQ, :x)
```

(Univariate polynomial ring in x over QQ, x)

julia> f = 1+x
x + 1

julia> parent(f//2)
Fraction field
  of univariate polynomial ring in x over QQ

julia> parent(f/2)
Univariate polynomial ring in x over QQ
```

The operator `//` should be understood as a constructor for formal fractions, not as a division operator.

If the parent is already a field (which means it is canonically isomorphic to its field of fractions), `//` coincides with exact division:

```julia

> julia> F101 = Native.GF(101)

Reviewer 2: ```suggestion
julia> F101 = GF(101)
```

- orders in number fields
- series rings

## [Subtle but important: `/` vs `//`](@id subtle_distinction_for_rings)

OSCAR distinguishes a number of different kinds of division. In particular, the operators `/` and `//` have distinct meanings.

> - `x / y` performs exact division within the current algebraic structure, if defined (and provides a convenient shorthand for `divexact`).

Reviewer 3: You don't state what `x`, `y` are. Also, I don't think the term "exact division" has any meaning for the average reader. How about this instead:
```suggestion
- If `x` and `y` are ring elements then `x / y` is a shorthand for `divexact(x,y)`  and performs division within their parent ring (raising an error if this is not possible).
```

- orders in number fields
- series rings

## [Subtle but important: `/` vs `//`](@id subtle_distinction_for_rings)

OSCAR distinguishes a number of different kinds of divisions. In particular, the operators `/` and `//` have distinct meanings.

Let `x` and `y` be elements of a ring.

- `x / y` is a shorthand for `divexact(x,y)` and performs division within their parent ring (raising an error if this is not possible).

> - `x // y` constructs a formal quotient, placing the result in a fraction-field parent.

Reviewer 3: I realize it's merged now, but in case you are interested in a follow-up: One could point out here that if the parent of x and y is a field, then x//y is the same as x/y after all ?

Example:

```julia
julia> R, x = polynomial_ring(QQ, :x)
(Univariate polynomial ring in x over QQ, x)

julia> f = 1+x
x + 1

julia> parent(f//2)
Fraction field
  of univariate polynomial ring in x over QQ

julia> parent(f/2)
Univariate polynomial ring in x over QQ
```

!!! note
    The above behavior applies to OSCAR types. For plain Julia numbers, `/`
    denotes floating-point division. Indeed, it is a common error to enter

>     `1/2` for the fraction 'one half' in Julia. We provide [more details on integer division in OSCAR.](@ref division_of_integers_in_OSCAR)

Reviewer 3: While this is strictly speaking correct, I wonder if it might not lead people away from trying e.g. `x/2` where `x isa RingElem` -- which does work. Perhaps worthwhile to mention this special case?
